src/proxyImplementationTemplate.ts

```diff
@@ -0,0 +1,27 @@
+export function proxyImplementationTemplate({
+  creatorFnName,
+  ownValueAlias,
+  useTranslateFn,
+}: {
+  creatorFnName: string;
+  ownValueAlias: string;
+  useTranslateFn: boolean;
+}) {
+  return `const ${creatorFnName} = <R extends string>(t: (...args: unknown[]) => R = (k) => k as R, prevKeys: string = ''): unknown => 
```

Comment:
are you adding the return type of `localeKeys` and also the translations in english?

Reply:
im not sure im getting ur question
moreover I did refactor implementation and covered it with tests

could u plz take a look at this piece again? or we can catchap directly